### PR TITLE
Decouple headtracker tests

### PR DESCRIPTION
**Pull request by dimriou**
Removed as many Core dependencies as we could from head tracker tests. Only package left is pgtest

### CODE REVIEW

File: core/chains/evm/log/helpers_test.go
Comment (amit-momin) [inline core/chains/evm/log/helpers_test.go:1]: I'm assuming `evm/log` is out-of-scope to completely decouple right? These were just changes needed after things moved out of `evmtest`?

**Review by amit-momin — DISMISSED** [review]

File: core/chains/evm/log/helpers_test.go
Comment (dimriou) [reply core/chains/evm/log/helpers_test.go:1]: The scope of this PR is only for Head Tracker. Besides that, log broadcaster is going to be deprecated so we just update whatever breaks during this change and nothing more.

**Comment by jmank88** [comment]
> Only package left is pgtest

Which part of the API is used? Some of that package can move to `chainlink-common` (or it may already have equivalents), but the `txdb` stuff is complex and worth investigating alternatives (which have already been exploring for core besides, without this extra complexity)

**Review by jmank88 — APPROVED** [review]

**Comment by dimriou** [comment]
> > Only package left is pgtest
> 
> Which part of the API is used? Some of that package can move to `chainlink-common` (or it may already have equivalents), but the `txdb` stuff is complex and worth investigating alternatives (which have already been exploring for core besides, without this extra complexity)

That's actually something I wanted to chat about at some point, but the biggest problem I see is that `orm_test` and similar files assume a db state that is currently located in Core. `pgtest` is used as a bootstraper for such tests so I'm not sure if there is an easy way to get around this even if we pull some code inside `chainlink-common`.
For example, in Head Tracker package, ther orm tests will try to execute the queries in a mock DB and they assume the state is already there but the db schema actually lives inside Core.

**Comment by jmank88** [comment]
> That's actually something I wanted to chat about at some point, but the biggest problem I see is that `orm_test` and similar files assume a db state that is currently located in Core. `pgtest` is used as a bootstraper for such tests so I'm not sure if there is an easy way to get around this even if we pull some code inside `chainlink-common`. For example, in Head Tracker package, ther orm tests will try to execute the queries in a mock DB and they assume the state is already there but the db schema actually lives inside Core.

@krehermann might have some opinions

**Review by amit-momin — APPROVED** [review]

**Pull request merged**